tls/s2n_server_new_session_ticket.c

```diff
@@ -253,3 +253,76 @@ int s2n_tls13_server_nst_write(struct s2n_connection *conn, struct s2n_stuffer *
 
     return S2N_SUCCESS;
 }
+
+/**
+ *= https://tools.ietf.org/rfc/rfc8446#section-4.6.1
+ *#     struct {
+ *#         uint32 ticket_lifetime;
+ *#         uint32 ticket_age_add;
+ *#         opaque ticket_nonce<0..255>;
+ *#         opaque ticket<1..2^16-1>;
+ *#         Extension extensions<0..2^16-2>;
+ *#      } NewSessionTicket;
+**/
+int s2n_tls13_server_nst_recv(struct s2n_connection *conn, struct s2n_stuffer *input)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(input);
+
+    POSIX_ENSURE(conn->mode == S2N_CLIENT, S2N_ERR_SAFETY);
+
+    if (conn->config->session_ticket_cb != NULL) {
+        uint32_t ticket_lifetime = 0;
+        POSIX_GUARD(s2n_stuffer_read_uint32(input, &ticket_lifetime));
+        /**
+         *= https://tools.ietf.org/rfc/rfc8446#section-4.6.1
+         *# The value of zero indicates that the
+         *# ticket should be discarded immediately.
+         */
+        if (ticket_lifetime == 0) {
+            return S2N_SUCCESS;
+        }
+        struct s2n_ticket_fields ticket_fields = { 0 };
+        POSIX_GUARD(s2n_stuffer_read_uint32(input, &ticket_fields.ticket_age_add));
+
+        uint8_t ticket_nonce_len = 0;
+        struct s2n_blob nonce = { 0 };
+        POSIX_GUARD(s2n_stuffer_read_uint8(input, &ticket_nonce_len));
+        uint8_t nonce_data[UINT8_MAX] = { 0 };
```

Comment:
Nit: why is writing the ticket_nonce_len sandwiched between your initializations for writing the nonce?